metecho/api/models.py

```diff
@@ -489,6 +533,99 @@ def get_collaborator(self, gh_uid: str) -> Optional[Dict[str, object]]:
             return None
 
 
+class ProjectDependency(HashIdMixin, TimestampsMixin):
+    name = StringField()
+    url = models.URLField()
+    recommended = models.BooleanField(default=False)
+
+    def __str__(self):
+        return self.name
+
+    class Meta:
+        verbose_name = _("project dependency")
+        verbose_name_plural = _("project dependencies")
+
+
+class GitHubOrganization(HashIdMixin, TimestampsMixin):
+    name = StringField()
+    login = StringField(unique=True, help_text="Organization's 'username' on GitHub")
```

Comment:
The mockup also states that one Org will be the "default option" -- so we probably need a way to set which Org is the main one?

I suppose if the current user doesn't have write access on the default org, then we don't pre-select any of the org options?

cc @davisagli

Reply:
Fixed in 9907fdddf3ac882bb70ffd339ed07648a69b3771

Reply:
I think having a default org is less important if we know which orgs the user is a member of and only provide those as options. (But doesn't it need to be an intersection of the orgs that the user is a member of and the orgs that the full-access github app has been installed on?)

Reply:
> I think having a default org is less important if we know which orgs the user is a member of and only provide those as options.

True. I'm thinking if there is one (and only one?) marked as "default" and it is an option for the given user, then we will pre-select it. Otherwise there will be no org pre-selected. Unless you'd prefer we get rid of the default boolean entirely?

> But doesn't it need to be an intersection of the orgs that the user is a member of and the orgs that the full-access github app has been installed on?

That's correct. @jerivas Right now, when a Project is added via the Django admin we check that Metecho has access to the repo, and display a ValidationError if it doesn't. Could we do that for GH orgs when they're created in the Django admin, to ensure that the full-access GH app has been installed?

Reply:
> I'm thinking if there is one (and only one?) marked as "default" and it is an option for the given user, then we will pre-select it. Otherwise there will be no org pre-selected. Unless you'd prefer we get rid of the default boolean entirely?

I'm fine with your suggestion, but I'm also having trouble thinking of a scenario where we would use the default boolean.

Reply:
Ah, okay -- I was under the impression that there was one org that would be used 95% of the time.

Reply:
Yes -- but if the full-access github app is only installed on that org then they won't have a choice anyway, and if it's installed on multiple orgs then I don't think there would be a reason to default to one over the other.

Reply:
I reverted the boolean field in 0593d4ffbdf1882d65e2376a49c28b23110d807d. @jgerigmeyer I couldn't find where we're checking for access to a repo in the Project admin (maybe you were thinking about the [branch and SHA detection](https://github.com/oddbird/Metecho/blob/0593d4ffbdf1882d65e2376a49c28b23110d807d/metecho/api/models.py#L397-L408)?). In any case I think we should be able to check for the permissions after saving on the admin. I'm assuming this would be a synchronous check, not offloaded to rq?

Reply:
@jerivas https://github.com/oddbird/Metecho/blob/0593d4ffbdf1882d65e2376a49c28b23110d807d/metecho/api/admin.py#L39-L48

Reply:
Oh, it was the form not the admin class 😅 Yeah, something like that should work for orgs too

Reply:
Added validation in 55581c57a76994af042b55600df9dc93c8b52c82